### DISCUSSION TITLE
How do you get mksurfdata_map on older CLM versions to work on Derecho?

### DISCUSSION THREAD

**Discussion by ekluzek** [ESCOMP/CTSM#2388] [Q&A] [answered]
With #2387 we will NOT be supporting mksurfdata_map for older release versions and ONLY supporting mksurfdata_esmf going forward from ctsm5.2.0 (not currently made yet as of writing).

So this leaves it up to the community who has interest in making this happen to figure how to do it. So we'd like you to post about your experiences here and what you need to do to get it to work. The general process should be similar for different versions, but might have some slight differences.

@lawrencepj1 can you post what you did in release-clm5.0.34 to get mksurfdata_map to work on Derecho?

**Comment by ekluzek** [comment]
Another way to handle this would be to use mksurfdata_esmf to create datasets (in versions at or after ctsm5.2.0) in older versions of the model (this will bring in answer changes to the datasets). This will NOT work out of the box, and you will have to apply changes to the CLM code in older versions to get this to work.

Is there an interest in this option? And would someone like to take a stab at what the steps would be?

**Reply by lawrencepj1** [reply, marked as answer]
Just as a quick note I have mksurf_datamap working on Derecho for CESM2.1.4

/glade/campaign/cesm/development/lmwg/landuse_source_tools/MksrfDataTools/cesm2_1_4_code/components/clm/tools/mksurfdata_map

Major change was to the netcdf libraries in Makefile.common in the src directory:

``` CPP
# Check for the netcdf library and include directories 
ifeq ($(LIB_NETCDF),$(null))
  LIB_NETCDF := /glade/u/apps/derecho/23.06/spack/opt/spack/netcdf/4.9.2/oneapi/2023.0.0/iijr/lib
endif

ifeq ($(INC_NETCDF),$(null))
  INC_NETCDF := /glade/u/apps/derecho/23.06/spack/opt/spack/netcdf/4.9.2/oneapi/2023.0.0/iijr/include
endif
```

Happy to talk more if people need help
Peter

**Reply by lawrencepj1** [reply]
Not sure what happened with formatting on the pasted text

**Reply by ekluzek** [reply]
@lawrencepj1 thanks for the notes. The formatting was due to markup options, so I made that section into CPP code, and now it looks great.
